src/Poker/Cards.hs

```diff
@@ -198,83 +197,83 @@ instance Bounded Card where
 allCards :: [Card]
 allCards = liftM2 Card allRanks allSuits
 
--- | 'Hand' represents a player's hole cards in a game of Texas Hold\'Em
-data Hand = MkHand !Card !Card
+-- | 'Hole' represents a player's hole cards in a game of Texas Hold\'Em
+data Hole = MkHole !Card !Card
   deriving (Eq, Ord, Show)
 
 -- TODO tests
-instance IsString Hand where
+instance IsString Hole where
   fromString = fromJust . parsePretty . T.pack
 
-{-# COMPLETE Hand #-}
+{-# COMPLETE Hole #-}
 
-pattern Hand :: Card -> Card -> Hand
-pattern Hand c1 c2 <- MkHand c1 c2
+pattern Hole :: Card -> Card -> Hole
+pattern Hole c1 c2 <- MkHole c1 c2
 
--- | Returns a 'Hand' if the incoming 'Card's are unique, else 'Nothing'.
--- Note that the internal representation of 'Hand' is normalised:
+-- | Returns a 'Hole' if the incoming 'Card's are unique, else 'Nothing'.
+-- Note that the internal representation of 'Hole' is normalised:
 --
 -- prop> mkHand c1 c2 == mkHand c2 c1
-mkHand :: Card -> Card -> Maybe Hand
-mkHand c1 c2
-  | c1 /= c2 = Just $ if c2 > c1 then MkHand c2 c1 else MkHand c1 c2
+mkHole :: Card -> Card -> Maybe Hole
+mkHole c1 c2
+  | c1 /= c2 = Just $ if c2 > c1 then MkHole c2 c1 else MkHole c1 c2
   | otherwise = Nothing
 
--- | Unsafely creates a new 'Hand'. The two input 'Card's are expected to be
+-- | Unsafely creates a new 'Hole'. The two input 'Card's are expected to be
 -- unique, and the first 'Card' should be less than the second 'Card' (as defined by
--- 'Ord'). See 'mkHand' for a safe way to create 'Hand's.
-unsafeMkHand :: Card -> Card -> Hand
-unsafeMkHand c1 c2 =
-  fromMaybe (terror $ "Cannot form a Hand from " <> prettyText (c1, c2)) $
-    mkHand c1 c2
+-- 'Ord'). See 'mkHole' for a safe way to create 'Hole's.
+unsafeMkHole :: Card -> Card -> Hole
+unsafeMkHole c1 c2 =
+  fromMaybe (terror $ "Cannot form a Hole from " <> prettyText (c1, c2)) $
+    mkHole c1 c2
 
--- | All possible Hold'Em poker 'Hand's
+-- | All possible Hold'Em poker 'Hole's
 --
 -- TODO add tests
-allHands :: [Hand]
-allHands = reverse $ do
+allHoles :: [Hole]
+allHoles = reverse $ do
   r1 <- enumerate
   r2 <- enumFrom r1
   (s1, s2) <-
     if r1 == r2
       then [(s1, s2) | s1 <- enumerate, s2 <- drop 1 (enumFrom s1)]
       else liftM2 (,) enumerate enumerate
-  pure $ unsafeMkHand (Card r1 s1) (Card r2 s2)
+  pure $ unsafeMkHole (Card r1 s1) (Card r2 s2)
 
-instance Enum Hand where
+instance Enum Hole where
```

Comment:
Without digging into it, this instance doesn't look safe. Can we ditch all the enum instances?

I like the convention of allWhatever, and enumerate can be replaced with direct usage of these lists..

Reply:
Agreed - I'll be removing this is future patches :)